src/contexts/walletconnect.tsx

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import WalletConnectClient, { CLIENT_EVENTS } from '@walletconnect/client'
+import { SessionTypes } from '@walletconnect/types'
```

Comment:
ping @LeeAlephium @polarker (not sure who added this change)

Reply:
@LeeAlephium I've been trying to upgrade the npm packages (after seeing many outdated packages when I run `npm outdated`, but after running `npm upgrade`, the build fails because the exports of `@walletconnect/types` have changed. Could you please have a look? Thanks!

Reply:
Maybe better for @LeeAlephium to investigate this. I was using `npm ci` for development and I don't have the time to investigate for now. I will help if I have more time later.

Reply:
I used the first link, and the rest was looking at source code. Their documentation is not really good right now because they're still technically in beta...

I can fix up the walletconnect/types usage, no problem :ok_hand:

Reply:
@nop33 the answer is `@walletconnect/client` pulls in `@walletconnect/types`

Reply:
> I used the first link

@LeeAlephium you probably mean the second link, cause that's the only one that uses `@walletconnect/client` and initializes the client the way we do. But that's confusing me a bit cause it's the docs for nodeJS.

Another place where I found a similar way to initialize the WC client was in the [react-wallet-v2](https://github.com/WalletConnect/web-examples/blob/main/wallets/react-wallet-v2/src/utils/WalletConnectUtil.ts) project of the web-examples, but there they use `@walletconnect/sign-client`.

So why did we go for `@walletconnect/client`?

Reply:
Oh wow, their documentation has actually changed significantly since a couple months ago...

Yes, used that project as a reference as well.

It's very possible the API changed significantly in 2 months since it _was_ and is still beta :upside_down_face: but damn that sucks... Anyway, if it compiles right now, it's all good. Can migrate once they finalize too.